sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/xml/XmlDataSource.scala

```diff
@@ -175,9 +178,24 @@ object MultiLineXmlDataSource extends XmlDataSource {
       file: PartitionedFile,
       parser: StaxXmlParser,
       requiredSchema: StructType): Iterator[InternalRow] = {
-    parser.parseStream(
-      CodecStreams.createInputStreamWithCloseResource(conf, file.toPath),
-      requiredSchema)
+    if (SQLConf.get.enableOptimizedXmlParser) {
```

Comment:
Will the optimized XML parser have any negative impacts? Why is it necessary to add a switch instead of directly replacing the old implementation with the optimized XML parser?